### PR TITLE
Update quickstart-java.md

### PR DESCRIPTION
**Pull request by robertwb**
I had to make this change for a clean build to work with the 0.6.0 source release.

### CODE REVIEW

File: src/get-started/quickstart-java.md [hunk 1]
```diff
@@ -35,7 +35,7 @@ $ mvn archetype:generate \
       -DarchetypeVersion=LATEST \
       -DgroupId=org.example \
       -DartifactId=word-count-beam \
-      -Dversion="0.1" \
+      -Dversion="0.6.0" \
```
Comment (davorbonaci) [inline src/get-started/quickstart-java.md]: This seems incorrect -- it is specifying the version of generated project, not the version of the artifact (or version of Beam).

Can you double check your experience here?

**Comment by robertwb** [comment]
I agree we shouldn't need this, and this probably isn't the right fix (but I'm not familiar enough with mvn to see what is).

```
curl -O https://dist.apache.org/repos/dist/dev/beam/0.6.0/apache-beam-0.6.0-source-release.zip
unzip apache-beam-0.6.0-source-release.zip
cd apache-beam-0.6.0
man compile
mvn archetype:generate \
      -DarchetypeRepository=https://repository.apache.org/content/groups/snapshots \
      -DarchetypeGroupId=org.apache.beam \
      -DarchetypeArtifactId=beam-sdks-java-maven-archetypes-examples \
      -DarchetypeVersion=LATEST \
      -DgroupId=org.example \
      -DartifactId=word-count-beam \
      -Dversion="0.1" \
      -Dpackage=org.apache.beam.examples \
      -DinteractiveMode=false
cd word-count-beam
mvn compile


[INFO] Scanning for projects...
[INFO] ------------------------------------------------------------------------
[INFO] Detecting the operating system and CPU architecture
[INFO] ------------------------------------------------------------------------
[INFO] os.detected.name: osx
[INFO] os.detected.arch: x86_64
[INFO] os.detected.classifier: osx-x86_64
[INFO]                                                                         
[INFO] ------------------------------------------------------------------------
[INFO] Building word-count-beam 0.1
[INFO] ------------------------------------------------------------------------
[WARNING] The POM for org.apache.beam:beam-sdks-common-runner-api:jar:0.1 is missing, no dependency information available
[WARNING] The POM for org.apache.beam:beam-sdks-java-core:jar:0.1 is missing, no dependency information available
[WARNING] The POM for org.apache.beam:beam-runners-core-construction-java:jar:0.1 is missing, no dependency information available
[WARNING] The POM for org.apache.beam:beam-runners-core-java:jar:0.1 is missing, no dependency information available
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 0.826 s
[INFO] Finished at: 2017-03-13T15:29:36-07:00
[INFO] Final Memory: 12M/309M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal on project word-count-beam: Could not resolve dependencies for project org.example:word-count-beam:jar:0.1: Failure to find org.apache.beam:beam-sdks-common-runner-api:jar:0.1 in https://repo.maven.apache.org/maven2 was cached in the local repository, resolution will not be reattempted until the update interval of central has elapsed or updates are forced -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/DependencyResolutionException
```

**Comment by davorbonaci** [comment]
You seem to be using the `0.6.0` release. It is not released yet -- it is not available in the Maven Central Repository, resulting in a failure to resolve dependencies.

During the release process, the candidate is staged in the Apache Staging Repository at https://repository.apache.org/content/repositories/staging/.

So, your build can be fixed right now by adding the following section to the pom.xml file:
```
  <repositories>
    <repository>
      <id>apache.staging</id>
      <name>Apache Development Snapshot Repository</name>
      <url>https://repository.apache.org/content/repositories/staging/</url>
      <releases>
        <enabled>true</enabled>
      </releases>
      <snapshots>
        <enabled>false</enabled>
      </snapshots>
    </repository>
  </repositories>
```

Currently, the POM is configured to look at Apache Snapshot repository (containing nightly builds) and Maven Central, but not Apache Staging (containing release candidates). Enabling snapshots is easy, and shouldn't have bad consequences unless somebody explicitly opts into -SNAPSHOTS. Enabling staging wasn't done -- unreleased code may get to users hands without an opt-in.

So, I'd say this is working as intended.

**Comment by davorbonaci** [comment]
Also, alternatively, had you done `mvn install` instead of `mvn compile` on the downloaded source, it might resolve everything from the local repository and it might just work.

**Comment by robertwb** [comment]
The fact that I didn't have a fully setup maven for a prerelease could very
well explain this, except that changing the version manually to 0.6.0 made
things work (!). I'm fine dropping this PR though if you think users won't
in general be impacted by this.

**Comment by davorbonaci** [comment]
Can't say for sure -- but seems suspicious. Is it possible that something else changed in your local repository in the meanwhile? If there's a reproduction, including removal of `~/.m2/` directory, I'd be happy to investigate.

**Comment by robertwb** [comment]
I'm going to close this as can't reproduce.

**Pull request closed without merging**